hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/expression/ExpressionVisitor.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.hive.expression;
+
+/**
+ * Visitor used to travers the expression.
+ */
+public interface ExpressionVisitor<T> {
+
+  T visitAnd(Expression left, Expression right);
+
+  T visitOr(Expression left, Expression right);
+
+  T visitBinaryComparator(Expression left, Expression.Operator operator, Expression right);
```

Comment:
@boneanxs appreciate the intent, but i'd suggest we do following:

1. Avoid creating specific instances of the operators like And, Or, etc. Instead let's just create `BinaryOperator` and then specify them using Operation type enum. This would allow us to avoid having 1 method for every operator and instead create just one method with a switch-case for operation type (this will help in implementation of [RFC-64](https://github.com/apache/hudi/pulls/alexeykudinkin), which i'm currently focusing on)
2. Also, let's make every class a top-level one (for ex, BinaryOperator, Literal, etc.)